File: 성민/레벨2/기능개발.js
function solution(progresses, speeds) {
  const answer = [];

  while (progresses.length) {
    let completeCnt = 0;
    for (let i = 0; i < progresses.length; i++) {
      if (progresses[i] < 100) {
        progresses[i] += speeds[i];
      }
    }
    while (progresses[0] >= 100) {
      progresses.shift();
      speeds.shift();
      completeCnt++;
    }
    if (completeCnt) {
      answer.push(completeCnt);
    }
  }
  return answer;
}

console.log(solution([99, 99, 99, 90, 90, 90], [1, 1, 1, 1, 1, 1]));

File: 종우/프로그래머스/기능개발.py
# 0104_1
# https://school.programmers.co.kr/learn/courses/30/lessons/42586

# 내 풀이
import math

def solution(progresses, speeds):
    answer = []
    dday = []
    for p, s in zip(progresses, speeds):
        pday = math.ceil((100-p)/s)
        dday.append(pday)
    
    nf = 1
    lt = dday.pop(0)
    for i in dday:
        if i <= lt:
            nf += 1
        else:
            answer.append(nf)
            nf = 1
            lt = i
    answer.append(nf)
    return answer

# 다른 풀이
def solution(progresses, speeds):
    Q=[]
    for p, s in zip(progresses, speeds):
        if len(Q)==0 or Q[-1][0]<-((p-100)//s):
            Q.append([-((p-100)//s),1])
        else:
            Q[-1][1]+=1
    return [q[1] for q in Q]